docker-entrypoint.sh

```diff
@@ -7,6 +7,7 @@ if [[ "$1" == apache2* ]] || [ "$1" == php-fpm ]; then
 \$cfg['blowfish_secret'] = '$(tr -dc 'a-zA-Z0-9~!@#$%^&*_()+}{?></";.,[]=-' < /dev/urandom | fold -w 32 | head -n 1)';
 EOT
     fi
+    chgrp www-data /etc/phpmyadmin/config.secret.inc.php
```

Comment:
This should probably be moved to the above if when the file is created, right ?

Reply:
no. testing it, the docker daemon resets the group each start. so it needs to be done each time the container starts.

Reply:
@williamdes What do you mean? Can we add this owner change to close a 5 years old issue? Don't get me wrong, but this will help a lot of people :)

Reply:
Sure, I am very interested too in closing this old bug
What bugs me is why can we not change the file owner at build time or create file time instead of repeatingly changing the group
In rootless mode this makes no sense to me: how a non root user would change the group of a file root owned??

Reply:
> Sure, I am very interested too in closing this old bug What bugs me is why can we not change the file owner at build time or create file time instead of repeatingly changing the group In rootless mode this makes no sense to me: how a non root user would change the group of a file root owned??

Security is exactly the point, not to change the file to be owned by www- user. In docker rootless mode, the file will be created inside the Container as running user/group, which would be root/root. This is safe, but cannot be read by phpmyadmin, as the user+group of apache would be www.
Setting the user to be www would mean giving the apache full rights to do whatever apache wants with that file, ALSO if the user would have NO RIGHTS TO WRITE to it (this is a often missunderstood feature of Linux).
The only way without using acls to give someone right to read an file without having the possibility to change it, is to not make the user own that file.
Also currently docker is a bit lazy of what groups and permissions are, so unfortunatelly it resets the group every time the container starts.
I don't see another way, if someone does, we could use it. But by now, rootless is only not working because of that.
So i would suppose to take that fix and make rootless work and after that if there are further problems - investigate seperately. only my two cents :)

Reply:
@J0WI do you have knowledge of other techniques?

Reply:
Currently, there is no mount an no config file. The file is part of running application, created inside the container itself at first start, giving an unique secret key for each instance.

This could be added, to maybe improve fpm/alpine compatibility in the future, but currently I don't see fpm images with phpmyadmin:
if [[ "$1" == apache2* ]] || [ "$1" = 'php-fpm' ]; then
...
group="${APACHE_RUN_GROUP:-www-data}"
...

As written above, user should stay root, group should he changed inside the container, so that for the host (running rootless, having umask 0007) the file will remain only read/writeable to the docker user and in the container file will only be changeable by root and only readable by Apache.

Reply:
> How do you define the umask ? Reading [moby/moby#19189](https://github.com/moby/moby/issues/19189) it seems to not be supported by Docker.
Well yes it is not by Docker itself. For my testing i have host facl set to:
```
# file: .
# owner: docker
# group: docker
user::rwx
group::--x
other::---
default:user::rwx
default:group::rwx
default:other::---
```
While those x are X more precisely (which make only directories X, not files), making the umask 0007 effectively.
My service has no umask set in docker.service, only my user has a ~./bashrc. Looks like docker will use the facl ones even in container itself.
So /etc/phpmyadmin/config.secret.inc.php will get 660 when created in entrypoint which is basically the wanted while secure default for new files as new files are usually created by the service that needs the file.
Only that file should be created by root and readable by www-data later.

Also after reading https://github.com/phpmyadmin/docker/issues/187#issuecomment-2033079756 i don't see any reason for not doing this pull.